catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/ScriptFragment.kt

```diff
@@ -0,0 +1,1009 @@
+            }
+            CATBLOCKS -> {
+            }
+        }
+        return true
+    }
+
+    override fun onPrepareActionMode(mode: ActionMode, menu: Menu): Boolean = false
+
+    override fun onActionItemClicked(mode: ActionMode, item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.confirm -> handleContextualAction()
+            else -> return false
+        }
+        return true
+    }
+
+    override fun onDestroyActionMode(mode: ActionMode) {
+        resetActionModeParameters()
+        adapter.clearSelection()
+        showBottomBar(activity)
+    }
+
+    private fun handleContextualAction() {
+        if (adapter.isEmpty) {
+            actionMode?.finish()
+        }
+        when (actionModeType) {
+            BACKPACK -> showNewScriptGroupAlert(adapter.selectedItems)
+            COPY -> copy(adapter.selectedItems)
+            DELETE -> showDeleteAlert(adapter.selectedItems)
+            COMMENT -> toggleComments(adapter.selectedItems)
+            NONE -> throw IllegalStateException("ActionModeType not set correctly")
+            CATBLOCKS -> {
+            }
+        }
+    }
+
+    private fun resetActionModeParameters() {
+        actionModeType = NONE
+        actionMode = null
+        adapter.setCheckBoxMode(BrickAdapter.NONE)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        val view = View.inflate(activity, R.layout.fragment_script, null)
+        listView = view.findViewById(android.R.id.list)
+        scriptFinder = view.findViewById(R.id.findview)
+
+        scriptFinder?.setOnResultFoundListener(object : OnResultFoundListener {
+            override fun onResultFound(
+                sceneIndex: Int,
+                spriteIndex: Int,
+                brickIndex: Int,
+                totalResults: Int,
+                textView: TextView?
+            ) {
+                val currentProject =
+                    projectManager.currentProject
+                val currentScene =
+                    currentProject.sceneList[sceneIndex]
+                val currentSprite =
+                    currentScene.spriteList[spriteIndex]
+                textView?.text = createActionBarTitle(
+                    currentProject,
+                    currentScene,
+                    currentSprite
+                )
+                projectManager.setCurrentSceneAndSprite(
+                    currentScene.name,
+                    currentSprite.name
+                )
+                adapter.updateItems(currentSprite)
+                adapter.notifyDataSetChanged()
+                listView?.smoothScrollToPosition(brickIndex)
+                highlightBrickAtIndex(brickIndex)
+                hideKeyboard()
+            }
+        })
+
+        val activity = activity as SpriteActivity?
+
+        scriptFinder?.setOnCloseListener(object : OnCloseListener {
```

Comment:
please convert to lambda.
```suggestion
        scriptFinder?.setOnCloseListener {
```

Reply:
![image](https://user-images.githubusercontent.com/46495964/125193382-c3d6a680-e269-11eb-907f-905745299617.png)